OpenEdXMobile/src/main/java/org/edx/mobile/view/adapters/DiscussionPostsSpinnerAdapter.java

```diff
@@ -70,51 +41,32 @@ public IconDrawable createIcon() {
     private final Spinner spinner;
 
     public DiscussionPostsSpinnerAdapter(@NonNull Spinner spinner,
-            @NonNull TextResourceProvider[] textResourceProviders,
-            @NonNull Icon icon) {
-        this(spinner, textResourceProviders,
-                new IconDrawableFactoryImpl(spinner.getContext(), icon));
-    }
-
-    public DiscussionPostsSpinnerAdapter(@NonNull Spinner spinner,
-            @NonNull TextResourceProvider[] textResourceProviders,
-            @NonNull IconDrawableFactory iconDrawableFactory) {
+                                         @NonNull TextResourceProvider[] textResourceProviders,
+                                         @DrawableRes int iconResId) {
         super(spinner.getContext(), R.layout.row_discussion_thread_dropdown, textResourceProviders);
-        this.iconDrawableFactory = iconDrawableFactory;
+        this.iconResId = iconResId;
         this.spinner = spinner;
     }
 
-    // The bounds are being set up manually instead of depending on the intrinsic
-    // dimensions because the constructed sort icon has non-square dimensions,
-    // which is not possible to set in IconDrawable.
-    @NonNull
-    private Drawable createIcon() {
-        Drawable icon = iconDrawableFactory.createIcon();
-        if (icon.getIntrinsicWidth() >= 0 && icon.getIntrinsicHeight() >= 0 &&
-                icon.getBounds().isEmpty()) {
-            icon.setBounds(0, 0, icon.getIntrinsicWidth(), icon.getIntrinsicHeight());
-        }
-        return icon;
-    }
-
     @Override
     @NonNull
     public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
         TextView textView = initTextView(position, convertView, parent);
         if (textView != convertView) {
-            TextViewCompat.setCompoundDrawablesRelative(
-                    textView, createIcon(), null, null, null);
+            UiUtils.INSTANCE.setTextViewDrawableStart(textView.getContext(), textView, iconResId,
+                    R.dimen.small_icon_size);
         }
         return textView;
     }
 
     @Override
     @NonNull
     public View getDropDownView(int position, @Nullable View convertView,
-            @NonNull ViewGroup parent) {
+                                @NonNull ViewGroup parent) {
         TextView textView = initTextView(position, convertView, parent);
         if (textView != convertView) {
-            Drawable icon = createIcon();
+            Drawable icon = UiUtils.INSTANCE.getDrawable(getContext(), iconResId,
```

Comment:
Can we getContext() from a view because in adapter there are chances for a null context?